runtime/src/accounts_index.rs

```diff
@@ -287,18 +287,28 @@ impl<T: IndexValue> ReadAccountMapEntry<T> {
 }
 
 pub struct PreAllocatedAccountMapEntry<T: IndexValue> {
-    entry: AccountMapEntry<T>,
+    entry: Option<AccountMapEntry<T>>,
+    raw: Option<(Slot, T)>,
 }
```

Comment:
It looks like `PreAllocatedAccountMapEntry` is either an `entry` or a `raw` (and one-and-only-one must be `Some`). Assuming that's the case, I think that's the perfect use for an `enum`. Maybe like this?

```
pub enum PreAllocatedAccountMapEntry<T: IndexValue> {
    Entry(AccountMapEntry<T>),
    Raw((Slot, T)),
}
```

Reply:
great idea